find_k only stores single eigenmode field info

It seems that the `find_k` function only records the first eigenmode's field into the ModeSolver object, even when it solves for multiple bands. Consequently, functions like `ModeSolver.get_efield()` can't access the different bands (or modes).

When trying to access multiple modes after a simulation, it throws the warning:
`Must have 1 <= band index <= num_bands (1)`

Here's a MWE:
```python
import meep as mp
from meep import mpb
import numpy as np
from matplotlib import pyplot as plt


Si   = mp.Medium(index=3.45)
SiO2 = mp.Medium(index=1.45)
wavelength = 1.55
omega      = 1/wavelength
w = 1    # Si width (um)
h = 0.5  # Si height (um)
geometry = [mp.Block(size=mp.Vector3(mp.inf, w, h), material=Si)]
resolution = 32  # pixels/um
sc_y = 2  # supercell width (um)
sc_z = 2  # supercell height (um)
geometry_lattice = mp.Lattice(size=mp.Vector3(0, sc_y, sc_z))

num_modes = 4
# Setup a simulation object
ms = mpb.ModeSolver(
    geometry_lattice = geometry_lattice,
    geometry         = geometry,
    resolution       = resolution,
    num_bands        = num_modes
)

# ------------------------------------------------- #
# Solve for all the modes at once, then plot
# ------------------------------------------------- #

k = ms.find_k(
    p              = mp.EVEN_Z,
    omega          = omega,
    band_min       = 1,
    band_max       = num_modes,
    korig_and_kdir = mp.Vector3(1,0,0),
    tol            = 1e-4,
    kmag_guess     = omega * 3.45,
    kmag_min       = omega * 0.1,
    kmag_max       = omega * 4
)
print(wavelength * np.array(k))
eps = ms.get_epsilon()
plt.figure(figsize=(8,4))
# Plot the E fields
for mode in range(num_modes):
    print('Current band: {}'.format(mode+1))
    E   = ms.get_efield(which_band=mode+1,bloch_phase=False)
    plt.subplot(1,num_modes,mode+1)
    plt.imshow(eps.transpose(), interpolation='spline36', cmap='binary')
    plt.imshow(np.abs(np.squeeze(E[:,:,0,1]).transpose())**2, cmap='RdBu', alpha=0.9)
    plt.axis('off')
    plt.title('$E_{{y{}}}$'.format(mode))
plt.tight_layout()
plt.savefig('errorFields.png')

# ------------------------------------------------- #
# Solve for all the modes one by one
# ------------------------------------------------- #

plt.figure(figsize=(8,4))
for mode in range(num_modes):
    k = ms.find_k(
        p              = mp.EVEN_Z,
        omega          = omega,
        band_min       = mode+1,
        band_max       = mode+1,
        korig_and_kdir = mp.Vector3(1,0,0),
        tol            = 1e-4,
        kmag_guess     = omega * 3.45,
        kmag_min       = omega * 0.1,
        kmag_max       = omega * 4
    )
    E   = ms.get_efield(which_band=mode+1,bloch_phase=False)
    plt.subplot(1,num_modes,mode+1)
    plt.imshow(eps.transpose(), interpolation='spline36', cmap='binary')
    plt.imshow(np.abs(np.squeeze(E[:,:,0,1]).transpose())**2, cmap='RdBu', alpha=0.9)
    plt.axis('off')
    plt.title('$E_{{y{}}}$'.format(mode))
plt.tight_layout()
plt.savefig('correctFields.png')

plt.show()
```

The problematic routine simply routines the fundamental mode after each call:

![errorFields](https://user-images.githubusercontent.com/24902086/57091321-244f1e00-6cc6-11e9-934b-dd59335af00e.png)

I can get the correct field profile if I iterate through the solver mode by mode (and only solve for that particular band):

![correctFields](https://user-images.githubusercontent.com/24902086/57091366-3fba2900-6cc6-11e9-94ec-e410bd364eed.png)

Hopefully this is just a small tweak in the python interface. I can look into it.

I can pass the `output_hfield()` or `output_efield()` band functions, and they store the field profiles in an h5 file. Is there a more elegant band function for caching the profiles in memory, and then pulling them after `find_k()` runs using the same 
```python
E   = ms.get_efield(which_band=mode+1,bloch_phase=False)
```

as before?

After #968, the example works just as expected:

```python
import meep as mp
from meep import mpb
import numpy as np
from matplotlib import pyplot as plt


Si   = mp.Medium(index=3.45)
SiO2 = mp.Medium(index=1.45)
wavelength = 1.55
omega      = 1/wavelength
w = 1    # Si width (um)
h = 0.5  # Si height (um)
geometry = [mp.Block(size=mp.Vector3(mp.inf, w, h), material=Si)]
resolution = 32  # pixels/um
sc_y = 2  # supercell width (um)
sc_z = 2  # supercell height (um)
geometry_lattice = mp.Lattice(size=mp.Vector3(0, sc_y, sc_z))

num_modes = 4
# Setup a simulation object
ms = mpb.ModeSolver(
    geometry_lattice = geometry_lattice,
    geometry         = geometry,
    resolution       = resolution,
    num_bands        = num_modes
)

# ------------------------------------------------- #
# Solve for all the modes at once, then plot
# ------------------------------------------------- #
E = []
store_fields = lambda ms_temp, mode: E.append(ms_temp.get_efield(which_band=mode,bloch_phase=False))    

k = ms.find_k(
    mp.EVEN_Z, # p
    omega, # omega
    1, # band_min
    num_modes, # band_max
    mp.Vector3(1,0,0), # korig_and_kdir
    1e-4, # tol
    omega * 3.45, # kmag_guess
    omega * 0.1, # kmag_min
    omega * 4, # kmag_max
    store_fields # band_funcs
)

eps = ms.get_epsilon()
plt.figure(figsize=(8,4))
# Plot the E fields
for mode in range(num_modes):
    print('Current band: {}'.format(mode+1))
    plt.subplot(1,num_modes,mode+1)
    plt.imshow(eps.transpose(), interpolation='spline36', cmap='binary')
    plt.imshow(np.abs(np.squeeze(E[mode][:,:,0,1]).transpose())**2, cmap='RdBu', alpha=0.9)
    plt.axis('off')
    plt.title('$E_{{y{}}}$'.format(mode))
plt.tight_layout()
plt.savefig('errorFields.png')

# ------------------------------------------------- #
# Solve for all the modes one by one
# ------------------------------------------------- #

plt.figure(figsize=(8,4))
for mode in range(num_modes):
    k = ms.find_k(
        p              = mp.EVEN_Z,
        omega          = omega,
        band_min       = mode+1,
        band_max       = mode+1,
        korig_and_kdir = mp.Vector3(1,0,0),
        tol            = 1e-4,
        kmag_guess     = omega * 3.45,
        kmag_min       = omega * 0.1,
        kmag_max       = omega * 4
    )
    E   = ms.get_efield(which_band=mode+1,bloch_phase=False)
    plt.subplot(1,num_modes,mode+1)
    plt.imshow(eps.transpose(), interpolation='spline36', cmap='binary')
    plt.imshow(np.abs(np.squeeze(E[:,:,0,1]).transpose())**2, cmap='RdBu', alpha=0.9)
    plt.axis('off')
    plt.title('$E_{{y{}}}$'.format(mode))
plt.tight_layout()
plt.savefig('correctFields.png')

plt.show()
```

Using band functions:

![errorFields](https://user-images.githubusercontent.com/24902086/61414536-73f7bb80-a8ab-11e9-9f42-aa3d3a3bad2d.png)

Manually iterating:

![correctFields](https://user-images.githubusercontent.com/24902086/61414556-7d812380-a8ab-11e9-9917-f69c51d0a704.png)

They agree!

--- a/python/solver.py
+++ b/python/solver.py
@@ -1040,7 +1040,7 @@ class ModeSolver(object):
             ks.append(mp.find_root_deriv(rootfun(b), tol, kmag_min, kmag_max, k0s[b - band_min]))
 
         if band_funcs:
-            for b, k in zip(range(band_max, band_max - nb, -1), reversed(ks)):
+            for b, k in zip(range(1, band_max +1), reversed(ks)):
                 self.num_bands = b
                 self.k_points = [korig + kdir1.scale(k)]
 

